### PR TITLE
format quarter cells

### PR DESCRIPTION
**Pull request by EyalMecilati**

### CODE REVIEW

File: cypress/integration/app/grid.spec.js [hunk 1]
```diff
@@ -344,5 +344,10 @@ context("grid", () => {
       cy.get("#submitCommodityAndCurrencyForm").click();
       checkGridValueByIndex(indexArr, ['7.33 (AUS$,m)', '-2.02 %', '5705 (lb,k)'], "test"); //npv,irr,Estimated FCF fields
     });
+
+    it("should show the user the right quorter and year (2022-04-01,2022-01-01)", function() {
```
Comment (alissaVrk) [inline cypress/integration/app/grid.spec.js:348]: this test should be a part of the single row - columns visibility with the rest of the non op stuff

File: src/logic/summery/formatValueByType.test.js [hunk 2]
```diff
@@ -107,4 +109,17 @@ describe("format values to user with the right sign for each value", () => {
     });
     expect(formatValue).toEqual("250 $");
   });
+
+  it("should get the right Quorter", () => {
+    const fieldDef = {
+      dataType: FieldDataTypes.Quorter,
+    };
+    forEach(dateToQuerter, (value, key) => {
```
Comment (alissaVrk) [inline src/logic/summery/formatValueByType.test.js:117]: you don't really need to go through all

File: src/utils/codeUtils.ts [hunk 3]
```diff
@@ -47,6 +47,14 @@ export const querterToDate: { [label: string]: string } = {
   "2027 Q3": moment(new Date("2027-07-01")).format("YYYY-MM-DD"),
   "2027 Q4": moment(new Date("2027-10-01")).format("YYYY-MM-DD"),
 };
+const formatDateToQuerter = (querterToDateObj: { [label: string]: string }) => {
+  let dateToQuerterObj = {};
+  forEach(querterToDateObj, (value, key) => {
+    dateToQuerterObj = { ...dateToQuerterObj, [value]: key };
```
Comment (alissaVrk) [inline src/utils/codeUtils.ts:53]: there is no need here for ...dateToQuerterObj
you can just do dateToQuerterObj[value] = key

**Review by alissaVrk — CHANGES_REQUESTED** [review]

File: src/utils/codeUtils.ts (hunk 3 above)
Comment (EyalMecilati) [reply src/utils/codeUtils.ts:53]: dateToQuerterObj[value] = key is not working i added Object.assign(dateToQuerterObj, {[value]:key })

Comment (alissaVrk) [reply src/utils/codeUtils.ts:53]: it should work.., what do you mean not working?

**Review by alissaVrk — DISMISSED** [review]

**Review by alissaVrk — APPROVED** [review]

**Pull request merged**